dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/sequence/FastaSequenceReader.java

```diff
@@ -0,0 +1,87 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.sequence;
+
+import htsjdk.samtools.SAMException;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.reference.ReferenceSequenceFile;
+import htsjdk.samtools.reference.ReferenceSequenceFileFactory;
+
+import java.nio.file.Path;
+
+/**
+ * Class used to read regions from a given FASTA file
+ */
+public class FastaSequenceReader {
+
+    private final ReferenceSequenceFile fastaSequenceFile;
+
+    private final SAMSequenceDictionary sequenceDictionary;
+
+    public FastaSequenceReader(Path fastaFile) {
+        fastaSequenceFile = ReferenceSequenceFileFactory.getReferenceSequenceFile(fastaFile, true);
+        sequenceDictionary = fastaSequenceFile.getSequenceDictionary();
+        if (sequenceDictionary == null || !fastaSequenceFile.isIndexed()) {
+            throw new IllegalArgumentException("A sequence dictionary file and a Fasta index file are required");
+        }
+    }
+
+    /**
+     * Get the sequence delimited by the give coordinates from a FASTA file
+     *
+     * @param contig Sequence contig or chromosome
+     * @param start  Sequence start coordinate in the contig
+     * @param end    Sequence end coordinate in the contig
+     * @return Sequence read from the FASTA file
+     * @throws IllegalArgumentException If the coordinates are not correct
+     * @throws ReadSequenceException    If there is an error reading the FASTA file
+     */
+    public String getSequence(String contig, long start,
+                              long end) throws IllegalArgumentException, ReadSequenceException {
+        checkArguments(contig, start, end);
+
+        try {
+            return fastaSequenceFile.getSubsequenceAt(contig, start, end).getBaseString();
+        } catch (SAMException e) {
+            throw new ReadSequenceException(e);
+        }
+    }
+
+    private void checkArguments(String contig, long start, long end) throws IllegalArgumentException {
+        if (end < start) {
+            throw new IllegalArgumentException("'end' must be greater or equal than 'start'");
+        } else if (start < 1) {
+            throw new IllegalArgumentException("'start' and 'end' must be positive numbers");
+        } else if (sequenceDictionary.getSequence(contig) == null) {
+            throw new IllegalArgumentException("Sequence " + contig + " not found in reference fasta file");
```

Comment:
This processor is being executed after the refseq to genbank map, so we should find GeneBank contigs in the variants. If we download the fasta files from Ensembl, I think that it will be OK. For the species not supported by Ensembl, I think they are not going to be imported into the data warehouse, so it shouldn' t matter.